### ISSUE TITLE
Invalid value of AF read; Flag ingerpretation

### ISSUE THREAD

**Issue by pstawinski** [lifeomic/spark-vcf#7]
Hi,
I have met surprising behaviour while loading this simple VCF with 2.1 version.
The source vcf looks like:
```
##fileformat=VCFv4.2
##INFO=<ID=AC,Number=A,Type=Integer,Description="Allele count in genotypes, for each ALT allele, in the same order as listed">
##INFO=<ID=AF,Number=A,Type=Float,Description="Allele Frequency, for each ALT allele, in the same order as listed">
##INFO=<ID=AN,Number=1,Type=Integer,Description="Total number of alleles in called genotypes">
##INFO=<ID=DB,Number=0,Type=Flag,Description="dbSNP Membership">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Total depth of called genotypes">
##INFO=<ID=MAJ,Number=1,Type=String,Description="Major allele is REF or ALT">
##contig=<ID=MT,length=16569>
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO
1       10109   rs376007522     A       T       22940.40        .       AC=86;AN=176;AF=0.4886;DP=17307;MAJ=REF;DB
1       10150   rs371194064     C       T       2276.70 .       AC=31;AN=176;AF=0.1761;DP=15279;MAJ=REF;DB
```

I read it using com.lifeomic.variants and below is what is read.
```
scala> val myVcf = spark.read.format("com.lifeomic.variants").option("use.info.map", "true").option("use.info.type", "true").load("....vcf")
scala> myVcf.show
+-----+-----+-----+-----+-----------+---+---+--------+------+---+---+-----+-----+----+----+
|chrom|  pos|start| stop|         id|ref|alt|    qual|filter| an|maj|   dp|   af|  ac|  db|
+-----+-----+-----+-----+-----------+---+---+--------+------+---+---+-----+-----+----+----+
|    1|10109|10108|10109|rs376007522|  A|  T|22940.40|     .|176|REF|17307|[0.0]|[86]|null|
|    1|10150|10149|10150|rs371194064|  C|  T| 2276.70|     .|176|REF|15279|[0.0]|[31]|null|
+-----+-----+-----+-----+-----------+---+---+--------+------+---+---+-----+-----+----+----+
```
What is suprising is:

1. The AF data is lost. It is set 0.0 without visible reason.
2. Flag DB is not interpreted correctly, I would expect db to have boolean value, while the DB data is lost completely here...

Thank you for your time, best wishes,
Piotr

**Comment by dmmiller612** [comment]
Yeah, I'll take a look. Thanks.

**Comment by dmmiller612** [comment]
My guess is the vcfs that we have looked at expect the info fields to be there. I'll try to get a patch up that tests with null info columns.

**Comment by dmmiller612** [comment]
Found the bug. A fix will be up momentarily and I can push it to maven central. Thanks for reporting it.

**Comment by dmmiller612** [comment]
Released version `0.3.0`. https://oss.sonatype.org/#nexus-search;quick~lifeomic . 

Some important notes that will be apart of the release notes. Type Info fields will now have a prefix attached to the key of `info_` due to the fact there were cases of collision in names with format fields.
